Automattic/jetpack
# AI Assistant: Add connection prompt when user is not connected to Jetpack

Fixes #30934

## Proposed changes:

* Adds a lib to check if the user is connected
* Adds a connection prompt if the user is not connected
* Disables multi-block panel if user is not connected

Known problems: Does not redirect the user back to the post after connection. ref p1685127123551239/1685125754.087739-slack-C054LN8RNVA

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Connect the site, but stop before connecting the user
* Add an AI Assistant block
* Check that there is a connection prompt and the block cannot be used
* Select two or more paragraphs
* Check that the sidebar does not display AI options
* Click on `Reconnect Jetpack` and follow the connection instructions
* Go back to the block editor
* Check that the assistant is unblocked and can be used normally

![2023-05-26_16-06-12](https://github.com/Automattic/jetpack/assets/8486249/319510be-0f69-4b77-bb0b-af13f85bbc4c)
![2023-05-26_16-06-24](https://github.com/Automattic/jetpack/assets/8486249/a2687983-01cc-4b95-b9ca-05da5e9788b5)
